### PR TITLE
fix: Add user_id check before error in event invoice API

### LINKED ISSUE #7402: Allow user to view their own event's invoices
Currently, users can view their account's invoices `/users/6/event-invoices`, but not able to see their event's invoices `/events/34/event-invoices`. Even though it is the same data. It is only possible for admin to do this. Allow users to access event invoice API for an event

### PR DESCRIPTION
**Pull request by manav1403**
Fixes #7402 

#### Short description of what this resolves:
added user_id check before Forbidden error is raised

#### Changes proposed in this pull request:

- Changes to if condition

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/event_invoices.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def query(self, view_kwargs):
         """
         user = current_user
         user_id = view_kwargs.get('user_id')
-        if user_id != user.id and not user.is_staff:
+        if user_id and user_id != user.id and not user.is_staff:
```
Comment (iamareebjamal) [inline app/api/event_invoices.py:33]: Good work. It should throw error if `/v1/event-invoices` is accessed

**Comment by manav1403** [comment]
@iamareebjamal pls review

File: app/api/event_invoices.py [hunk 2]
```diff
@@ -30,6 +30,8 @@ def query(self, view_kwargs):
         """
         user = current_user
         user_id = view_kwargs.get('user_id')
+        if '/v1/event-invoices' in request.url and not user.is_staff:
```
Comment (iamareebjamal) [inline app/api/event_invoices.py:33]: This is not how we do it

Comment (manav1403) [reply app/api/event_invoices.py:33]: can you guide me?

Comment (iamareebjamal) [reply app/api/event_invoices.py:33]: Check view_args

Comment (manav1403) [reply app/api/event_invoices.py]: Is the way I opted now is correct or still go for args?

File: app/api/event_invoices.py [hunk 3]
```diff
@@ -30,7 +30,11 @@ def query(self, view_kwargs):
         """
         user = current_user
         user_id = view_kwargs.get('user_id')
-        if user_id != user.id and not user.is_staff:
+        params = user_id or view_kwargs.get('event_id') \
+            or view_kwargs.get('event_identifier')
+        if not params and not user.is_staff:
+            raise ForbiddenError({'source': ''}, 'Admin access is required')
+        if user_id and user_id != user.id and not user.is_staff:
```
Comment (iamareebjamal) [inline app/api/event_invoices.py:37]: Both the conditions can be combined

File: app/api/event_invoices.py [hunk 4]
```diff
@@ -30,7 +30,9 @@ def query(self, view_kwargs):
         """
         user = current_user
         user_id = view_kwargs.get('user_id')
-        if '/v1/event-invoices' in request.url and not user.is_staff:
+        params = user_id or view_kwargs.get('event_id') \
+            or view_kwargs.get('event_identifier')
+        if not params and not user.is_staff:
```
Comment (iamareebjamal) [inline app/api/event_invoices.py:35]: I think checking for an empty dict is much easier

File: app/api/event_invoices.py [hunk 5]
```diff
@@ -30,11 +30,9 @@ def query(self, view_kwargs):
         """
         user = current_user
         user_id = view_kwargs.get('user_id')
-        params = user_id or view_kwargs.get('event_id') \
-            or view_kwargs.get('event_identifier')
-        if not params and not user.is_staff:
-            raise ForbiddenError({'source': ''}, 'Admin access is required')
-        if user_id and user_id != user.id and not user.is_staff:
+        check = user_id and user_id != user.id \
```
Comment (iamareebjamal) [inline app/api/event_invoices.py:33]:
```suggestion
        forbidden = (user_id and user_id != user.id) \
```

File: app/api/event_invoices.py [hunk 6]
```diff
@@ -30,11 +30,9 @@ def query(self, view_kwargs):
...
+            or not view_kwargs
+        if check and not user.is_staff:
```
Comment (iamareebjamal) [inline app/api/event_invoices.py:35]:
```suggestion
        if forbidden and not user.is_staff:
```

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**